dozer-spring-boot-auto-configuration/dozer-spring-boot-autoconfigure/src/main/java/com/github/dozermapper/springboot/autoconfigure/DozerConfigurationProperties.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2005-2017 Dozer Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.dozermapper.springboot.autoconfigure;
+
+import java.util.Arrays;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.core.io.Resource;
+
+
+/**
+ * Dozer configuration properties.
+ */
+@ConfigurationProperties(prefix = "dozer")
+public class DozerConfigurationProperties {
+
+    private Resource[] mappingFiles = new Resource[] {};
```

Comment:
It would be nice to have a javadoc on this field (and maybe event skipping javadocs for getter and setter, to avoid unnecessary duplication).

The reason to have a javadoc on the field is to generate a better [meta-data file](https://docs.spring.io/spring-boot/docs/current/reference/html/configuration-metadata.html) to help IDE to provide richer support for these properties, compare
```
"properties": [
    {
      "sourceType": "com.github.dozermapper.springboot.autoconfigure.DozerConfigurationProperties",
      "defaultValue": [],
      "name": "dozer.mapping-files",
      "type": "org.springframework.core.io.Resource[]"
    }
  ]
```
to
```
"properties": [
    {
      "sourceType": "com.github.dozermapper.springboot.autoconfigure.DozerConfigurationProperties",
      "defaultValue": [],
      "name": "dozer.mapping-files",
      "description": "Mapping files configuration.\n For example <code>classpath:*.dozer.xml<\/code>.",
      "type": "org.springframework.core.io.Resource[]"
    }
  ]
```

Reply:
I agree. But to avoid unnecessary duplication `maven-checkstyle-plugin` configuration should be updated. Are you fine with that?

Reply:
In general I do not see any problems in tuning tools to suite the project needs. In this particular case I would recommend to use `@SuppressWarnings("checkstyle:xxx")` in code + enabling [SuppressWarningsFilter](http://checkstyle.sourceforge.net/config_filters.html#SuppressWarningsFilter) in CheckStyle config.